lib/bundix.rb

```diff
@@ -127,7 +127,7 @@ def build_depcache(lock)
     end
 
     lock.specs.each do |spec|
-      dep_cache[spec.name] ||= Dependency.new(spec.name, nil, {})
+      dep_cache[spec.name] ||= Dependency.new(spec.name, nil, { "group" => [] })
```

Comment:
This loop adds everything in the lockfile that isn't listed in the gemfile so technically we don't know which groups these gems are supposed to be.

If `group` is not specified, the constructor defaults to `default` which is not what we want as we want to resolve groups based on the dependency tree. So we explicitly set the `group` to empty which we're going to fill in later.